### PR TITLE
Stale column headers

### LINKED ISSUE #390: URL Query params do not update when changing direction
There's an issue with the URL query params - you cannot change direction back to a previously selected direction for a given column

To replicate
- Click on the currently sorted column: the sort direction changes
- Click on the column again: no change in direction and the query search parameters don't change

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #390 

🤦‍♂️ Hoo boy. glad that no one saw me do that

## Tests

- [ ] Click on a header: column should sort
- [ ] Click on same header: column should sort in reverse direction
- [ ] Try a few more times: 🤷

### CODE REVIEW

File: client/packages/common/src/hooks/useUrlQuery/useUrlQuery.ts [hunk 1]
```diff
@@ -16,17 +16,14 @@ export const useUrlQuery = ({ skipParse = [] }: useUrlQueryProps = {}) => {
   const updateQuery = (values: UrlQueryObject, overwrite = false) => {
     const newQueryObject = overwrite
       ? {}
-      : { ...parseSearchParams(searchParams, skipParse) };
+      : Object.fromEntries(searchParams.entries());
```
Comment (mark-prins) [inline client/packages/common/src/hooks/useUrlQuery/useUrlQuery.ts:19]: a quick tidy while I'm here. No need to create a parsed object and then convert all values to strings again.

File: client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts [hunk 2]
```diff
@@ -45,10 +45,7 @@ export const useUrlQueryParams = ({
     if (sort !== currentSort) {
       updateQuery({ sort, dir: '', page: '' });
     } else {
-      const dir =
-        column.sortBy?.direction === 'asc' || !column.sortBy?.direction
-          ? 'desc'
-          : '';
+      const dir = column.sortBy?.direction === 'desc' ? '' : 'desc';
```
Comment (mark-prins) [inline client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts:48]: same? but less code.. 
The expected values are `'desc'` `'asc'` `undefined` and we just want to toggle

File: client/packages/common/src/ui/layout/tables/DataTable.tsx [hunk 3]
```diff
@@ -39,6 +39,10 @@ export const DataTableComponent = <T extends RecordWithId>({
   const { setRows, setDisabledRows, setFocus } = useTableStore();
   const [clickFocusedRow, setClickFocusedRow] = useState(false);
   const [displayColumns, setDisplayColumns] = useState(columns);
+  const columnsToDisplay = columns.filter(c =>
```
Comment (mark-prins) [inline client/packages/common/src/ui/layout/tables/DataTable.tsx:42]: 🙄  use the actual column, not the stale version from the state

File: client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts (hunk 2 above)
Comment (andreievg) [reply client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts:48]: Yep, that does look better !

File: client/packages/common/src/hooks/useUrlQuery/useUrlQuery.ts (hunk 1 above)
Comment (andreievg) [reply client/packages/common/src/hooks/useUrlQuery/useUrlQuery.ts:19]: that also looks much better thanks

**Review by andreievg — APPROVED** [review]
Looks good thanks.

**Pull request merged**